subgraphs/truefi/src/entities/price.ts

```diff
@@ -0,0 +1,73 @@
+import { Address, BigDecimal, BigInt, log } from "@graphprotocol/graph-ts";
+import { Token } from "../../generated/schema";
+import { ChainlinkTruUsdcOracle } from "../../generated/LinearTrueDistributor/ChainlinkTruUsdcOracle";
+import {
+  BIGDECIMAL_ZERO,
+  BIGINT_ZERO,
+  STABLECOIN_USDT_ADDRESS,
+  STABLECOIN_USDC_ADDRESS,
+  STABLECOIN_BUSD_ADDRESS,
+  STABLECOIN_TUSD_ADDRESS,
+  TRU_ORACLE_ADDRESS,
+  BIGINT_TEN,
+  BIGDECIMAL_ONE,
+  BIGDECIMAL_TEN,
+} from "../utils/constants";
+import { bigIntToBigDecimal } from "../utils/numbers";
+import { getTokenById } from "./token";
+
+export function getTruPrice(decimals: i32): BigDecimal {
+  const priceOracleAddress = Address.fromString(TRU_ORACLE_ADDRESS);
+  const priceOracle = ChainlinkTruUsdcOracle.bind(priceOracleAddress);
+  const price = priceOracle.try_getLatestTruPrice();
+  if (price.reverted) {
+    log.error(
+      "Failed to fetch latest Tru price, contract call reverted. Price oracle: {}",
+      [priceOracleAddress.toHexString()]
+    );
+    return BIGDECIMAL_ZERO;
+  }
+
+  return bigIntToBigDecimal(price.value, decimals);
+}
+
+export function amountInUSD(amount: BigInt, token: Token): BigDecimal {
+  if (amount == BIGINT_ZERO) {
+    return BIGDECIMAL_ZERO;
+  }
+  if (token.underlyingAsset) {
+    return amountInUSD(amount, getTokenById(token.underlyingAsset!));
+  }
+
+  if (
+    token.id == STABLECOIN_USDT_ADDRESS ||
+    token.id == STABLECOIN_USDC_ADDRESS ||
+    token.id == STABLECOIN_BUSD_ADDRESS ||
+    token.id == STABLECOIN_TUSD_ADDRESS
+  ) {
+    return bigIntToBigDecimal(amount, token.decimals);
+  }
+  return bigIntToBigDecimal(amount.div(BIGINT_TEN), token.decimals);
+}
+
+export function getInputTokenPrice(token: Token): BigDecimal {
+  const price = BIGDECIMAL_ONE;
```

Comment:
Const is divided by 10 but not save into same const again so it should not have problems. Otherwise the compiler will give back errors. Fix the oracle problem in d99eb453ec77afb674d3a370e1d0c781708ed260